hscommon/hsenum/tile.go

```diff
@@ -23,3 +23,40 @@ const (
 	TileLowerWallsEquivalentToRightLeftNorthCornerWall
 	TileLowerWallsEquivalentToSouthCornerwall
 )
+
+// GetTileTypeString returns string of tile type
+// nolint:gocyclo // can't reduce
+func GetTileTypeString(t int32) string {
```

Comment:
we could probably do `type tileType int32` and implement the stringer interface. then, this func would become a method `String() string` of the `tileType` type. What is currently here is fine, but implementing the stringer interface is probably less taboo.

Also, the reason for the nolint directive is not quite right, we dont even need a nolint directive if a `map[int32]string` were declared inside of the func and you just returned `stringMap[t]`

a combination of both points above:
```golang
type tileType int32

func (t tileType) String() string {
	strings := map[tileType]string{
		TileFloor: "floor",
		TileSpecialTile2: "special",
		TileSpecialTile2: "special",
		TileShadow: "shadow",
		TileTree: "wall/object",
		TileRoof: "roof",
		TileLeftWall: "Left Wall",
		TileRightWall: "Upper Wall",
		TileRightPartOfNorthCornerWall: "Upper part of an Upper-Left corner",
		TileLeftPartOfNorthCornerWall: "Left part of an Upper-Left corner",
		TileLeftEndWall: "Upper-Right corner",
		TileRightEndWall: "Lower-Left corner",
		TileSouthCornerWall: "Lower-Right corner",
		TileLeftWallWithDoor: "Left Wall with Door object, but not always",
		TileRightWallWithDoor: "Upper Wall with Door object, but not always",
	}

	str, found := strings[t]

	if !found {
		str = "lower wall ?"
	}

	return str
}
```

Reply:
IMO it should be method of `d2enum.TileType`

Reply:
Agreed

Reply:
OpenDiablo2/HellSpawner#1089